lms/djangoapps/teams/static/teams/js/views/teams_tab.js

```diff
@@ -365,7 +365,7 @@
                         viewWithHeader = this.createViewWithHeader({
                             subject: topic,
                             mainView: teamsView,
-                            headerActionsView: null, // TODO: add back SearchFieldView when search is enabled
+                            headerActionsView: SearchFieldView,
```

Comment:
I believe this is the problem. You are including the class ``SearchFieldView``, not the instance ``searchFieldView``. I'm surprised that the search tests work with this bug.